### PR TITLE
api changes to allow post and get icfp prepopulated data plan

### PR DESCRIPTION
**Pull request by J05h-L**
### Context
part of the icfp journey - this is for the prepopulated data page
Story - [AB#191578](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/191578)
Task - [AB#191733](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/191733)

### Change proposed in this pull request
this PR adds:
- changes records for req/res/db to allow post and get of the financial plan

outstanding to complete ticket:
- display financial plan when it exists for the page
- validation on web app

### Guidance to review 
N/A

### Checklist
- [x] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [x] Your code builds clean without any errors or warnings
- [x] You have run all unit/integration tests and they pass
- [x] Your branch has been rebased onto main
- [x] You have tested by running locally

### CODE REVIEW

File: platform/src/abstractions/EducationBenchmarking.Platform.Domain/Requests/FinancialPlanRequest.cs [hunk 1]
```diff
@@ -3,4 +3,10 @@ namespace EducationBenchmarking.Platform.Domain.Requests;
 public record FinancialPlanRequest
 {
     public string? User { get; set; }
+    public bool UseFigures { get; set; }
+    public string TotalIncome { get; set; }
```
Comment (jrabbott) [inline platform/src/abstractions/EducationBenchmarking.Platform.Domain/Requests/FinancialPlanRequest.cs:7]: These should be nullable decimal

File: platform/src/abstractions/EducationBenchmarking.Platform.Domain/Responses/FinancialPlan.cs [hunk 2]
```diff
@@ -12,7 +12,13 @@ public record FinancialPlan
     public string? UpdatedBy { get; set; }
     public string? CreatedBy { get; set; }
     public int Version { get; set; }
-    
+    public string? TotalIncome { get; set; }
```
Comment (jrabbott) [inline platform/src/abstractions/EducationBenchmarking.Platform.Domain/Responses/FinancialPlan.cs:15]: These should be nullable decimal

**Review by jrabbott — CHANGES_REQUESTED** [review]

File: platform/src/abstractions/EducationBenchmarking.Platform.Domain/Requests/FinancialPlanRequest.cs (hunk 1 above)
Comment (J05h-L) [reply platform/src/abstractions/EducationBenchmarking.Platform.Domain/Requests/FinancialPlanRequest.cs]: pushed up changes

File: platform/src/abstractions/EducationBenchmarking.Platform.Domain/Responses/FinancialPlan.cs (hunk 2 above)
Comment (J05h-L) [reply platform/src/abstractions/EducationBenchmarking.Platform.Domain/Responses/FinancialPlan.cs]: pushed up changes

**Review by jrabbott — APPROVED** [review]

**Pull request merged**